storage: avoid spurious nodeDialer logs

tbg opened the pull request:
breaker.Success() retrieves its count from an underlying windowed
counter that resets every 10 seconds. As a result, spurious "established
connection to nX" messages would pop up every 10s.

It turns out it's annoyingly difficult to properly gauge whether an
established connection is the first of its kind. In trying to do so, I
became unconvinced that this is even worth it, so I ended up removing
it. I similarly simplified the logging in the unsuccessful case (though
ConsecutiveFailures is actually not a windowed counter and presumably
worked correctly).

I also downgraded the message in the error case to Infof, as it's
expected to see it when nodes are restarted (which is a routine
operation and nothing to worry about). We want to keep the Warning level
for true warnings.

Release note: None

tbg committed:
storage: avoid spurious nodeDialer logs

breaker.Success() retrieves its count from an underlying windowed
counter that resets every 10 seconds. As a result, spurious "established
connection to nX" messages would pop up every 10s.

It turns out it's annoyingly difficult to properly gauge whether an
established connection is the first of its kind. In trying to do so, I
became unconvinced that this is even worth it, so I ended up removing
it. I similarly simplified the logging in the unsuccessful case (though
ConsecutiveFailures is actually not a windowed counter and presumably
worked correctly).

I also downgraded the message in the error case to Infof, as it's
expected to see it when nodes are restarted (which is a routine
operation and nothing to worry about). We want to keep the Warning level
for true warnings.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/31017)
<!-- Reviewable:end -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31017#-:-LO4MEpR6s4B3nx7lGDZ:bnfp4nl)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/rpc/nodedialer/nodedialer.go, line 92 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31017#-LO4MAXlD-PB2EC3aoMB:-LO4MAXlD-PB2EC3aoMC:b-qatbrv) ([raw file](https://github.com/cockroachdb/cockroach/blob/9c8037d906633d0c952fe07f49eb792592fe5a8f/pkg/rpc/nodedialer/nodedialer.go#L92)):*
> ```Go
> 				}
> 			} else {
> 				log.Infof(ctx, "connection to n%d established", nodeID)
> ```

You're ok with losing this log?


<!-- Sent from Reviewable.io -->

petermattis reviewed (commented):
Hold up for a second. Did I make this worse by calling `breaker.Reset` when the breaker was open?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31017#-:-LO4O71a-_qkioLDmJUv:b-k07gbe)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

petermattis reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31017)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[pkg/rpc/nodedialer/nodedialer.go, line 152 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31017#-LO4OQf8EvViDwwmMFPy:-LO4OQf8EvViDwwmMFPz:b-hr5hx9) ([raw file](https://github.com/cockroachdb/cockroach/blob/0cd78b2ca3579c431370e0721c25a48160483ecd/pkg/rpc/nodedialer/nodedialer.go#L152)):*
> ```Go
> 	// RPCs fail when dial fails due to an open breaker. Reset the breaker here
> 	// as a stop-gap before the reconciliation occurs.
> 	n.getBreaker(nodeID).Reset()
> ```

I'm wondering if this should be something like:

```
if breaker := n.getBreaker(nodeID); breaker.Tripped() {
  breaker.Reset()
}
```


<!-- Sent from Reviewable.io -->

tbg reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31017)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[pkg/rpc/nodedialer/nodedialer.go, line 92 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31017#-LO4MAXlD-PB2EC3aoMB:-LO4YqzaEVc3I8c4ls_S:b-87u5yy) ([raw file](https://github.com/cockroachdb/cockroach/blob/9c8037d906633d0c952fe07f49eb792592fe5a8f/pkg/rpc/nodedialer/nodedialer.go#L92)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

You're ok with losing this log?
</blockquote></details>

I am.
___
*[pkg/rpc/nodedialer/nodedialer.go, line 152 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31017#-LO4OQf8EvViDwwmMFPy:-LO4Yy3bEaH4n9ruxYqh:bupsb7k) ([raw file](https://github.com/cockroachdb/cockroach/blob/0cd78b2ca3579c431370e0721c25a48160483ecd/pkg/rpc/nodedialer/nodedialer.go#L152)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

I'm wondering if this should be something like:

```
if breaker := n.getBreaker(nodeID); breaker.Tripped() {
  breaker.Reset()
}
```
</blockquote></details>

That sounds a little more than what you were trying to do, but the `Successes()` thing was just flawed. Though now that I think about it, yeah this is probably happening again now due to your change. Since the window counter has multiple windows, you'd need no request for all windows and only then would you get a spurious log message.

I still think this PR is a good change. Relying on that breaker for this logging is a bad idea.


<!-- Sent from Reviewable.io -->

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31017#-:-LO4aQC26fzuVkn1uokf:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/rpc/nodedialer/nodedialer.go, line 152 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31017#-LO4OQf8EvViDwwmMFPy:-LO4_fui0r1YFpP3hyOz:b-te2wk8) ([raw file](https://github.com/cockroachdb/cockroach/blob/0cd78b2ca3579c431370e0721c25a48160483ecd/pkg/rpc/nodedialer/nodedialer.go#L152)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

That sounds a little more than what you were trying to do, but the `Successes()` thing was just flawed. Though now that I think about it, yeah this is probably happening again now due to your change. Since the window counter has multiple windows, you'd need no request for all windows and only then would you get a spurious log message.

I still think this PR is a good change. Relying on that breaker for this logging is a bad idea.
</blockquote></details>

Ack. Can you tack my suggested change on to this PR as well? Seems good to not be calling `Reset` unless the breaker was `Tripped`.


<!-- Sent from Reviewable.io -->

tbg committed:
nodedialer: Reset() breaker only when it is tripped

This is slightly better since it only resets the breaker state (and with
it the underlying counters) when it's tripped.

This shouldn't result in any external changes but made the problem fixed
in #31017 much more visible than it would've otherwise been.

Release note: None

tbg reviewed (commented):
TFTRs!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31017#-:-LO4cQNHA1CeHiYv3MpG:bz5mair)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 2 stale)

___
*[pkg/rpc/nodedialer/nodedialer.go, line 92 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31017#-LO4MAXlD-PB2EC3aoMB:-LO4cPpKEst2qCNQL5k_:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9c8037d906633d0c952fe07f49eb792592fe5a8f/pkg/rpc/nodedialer/nodedialer.go#L92)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

I am.
</blockquote></details>

Done.
___
*[pkg/rpc/nodedialer/nodedialer.go, line 152 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/31017#-LO4OQf8EvViDwwmMFPy:-LO4cOvg4MvWDRssFVkz:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0cd78b2ca3579c431370e0721c25a48160483ecd/pkg/rpc/nodedialer/nodedialer.go#L152)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Ack. Can you tack my suggested change on to this PR as well? Seems good to not be calling `Reset` unless the breaker was `Tripped`.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

tschottdorf commented:
bors r=petermattis,nvanbenschoten
